core/src/test/scala/com/rackspace/com/papi/components/checker/wadl/BaseCheckerSpec.scala

```diff
@@ -428,28 +429,28 @@ class BaseCheckerSpec extends BaseWADLSpec {
 
     if (step_funs.isEmpty) throw new TestFailedException("Path assertion should contain at least one step!",4)
 
-    def followPath(step : NodeSeq, nextSteps : Seq[NodeSeq]) : Boolean = {
-      if (nextSteps.isEmpty) return true
-      val next = nextSteps(0)
-      step.foreach(n =>
-        (n \ "@next").text.split(" ").foreach(m => next.foreach (o =>
-          if ((o \ "@id").text == m) {
-            if (followPath (next, nextSteps.drop(1))) return true
-          })))
-      false
-    }
 
-    val steps = step_funs.map (n => n(checker))
+    def getNext(step : Node) : NodeSeq = {
+      (step \ "@next").text.split(" ").map(id => { (checker \\ "step").filter (ns => { (ns \ "@id").text == id}) }).fold(NodeSeq.Empty)(_ ++ _)
+    }
 
-    //
-    //  Check to make sure all the steps are available...
-    //
-    steps.foreach (s => if (s.isEmpty) throw new TestFailedException("A step in the path could not be found", 4))
+    def followPath(steps : NodeSeq, stepFuns : List[(NodeSeq) => NodeSeq]) : Boolean = stepFuns match {
+      case Nil => true
+      case stepFun :: nextFuns =>
+        val matchSteps = stepFun(steps)
+        logger.trace ("Match Steps "+matchSteps)
+        if (matchSteps.isEmpty) return false
```

Comment:
It is recommended that this `if`/`return`/`more stuff` should be an `if`/`else` block. In Odersky's _"Programming in Scala"_ it states:
> "The recommended style for methods is in fact to avoid having explicit, and especially multiple, return statements." [1]
1. http://www.artima.com/pins1ed/classes-and-objects.html